Layouts-spotify-Assingment-project

mugayamadox opened the pull request:
This is my initial submission for the spotify project.

mugayamadox committed:
IMPLEMENT:  added navigation and landing

Added a navigation and landing content

Navigation:
   Logo
   List of menu

Landing content:
    Landing image
    Landing text

Files affected:
    index.html
     styles/style.css

mugayamadox committed:
LAYOUT: worked on the feature layout

BRIEF:
Worked on the feature section layout

ADDED:
Icons, and separated each feature in a separate div.

FILES AFFECTED:
index.html
styles/style.css

mugayamadox committed:
ADDED: Section 2 and 3

WORKED ON:
Section 2 and 3

ADDED:
Spotify features and discover sections

FILES AFFECTED
1. index.html
2. style.css

edwinmp reviewed (changes requested):
Great job on this.
Now, lets get you out of the ID habit. You next review request should not have a single ID that you can't explain as necessary.
Also give your body some styles so it's not affected by default browser styles e.g. my Chrome & Firefox give your body a default margin of 8px.

mugayamadox commented:
> Great job on this.
> Now, lets get you out of the ID habit. You next review request should not have a single ID that you can't explain as necessary.
> Also give your body some styles so it's not affected by default browser styles e.g. my Chrome & Firefox give your body a default margin of 8px.

Hi there Mr. Edwin

So on this issue of IDs. 
Honestly as far as my layout  implementation skills are concerned, 
Each ID I used was necessary.
Though I am sure that with further knowledge of layout techniques maybe one can use fewer IDs. And avoid the clutter in my submission.
 
Inorder to adjust, I need your recommendations on how I can improve on my DRY abilities and suggestions on a sample of what would be a better implementation. With that, I guess I fill the gaps.
Thanks

edwinmp commented:
> Inorder to adjust, I need your recommendations on how I can improve on my DRY abilities and suggestions on a sample of what would be a better implementation. With that, I guess I fill the gaps.

Simple. Instead of an ID, use a CSS class. There's no mysterious knowledge to it. Your ID even have proper names that perfectly fit classes, so just use classes.

So, instead of `#wrapper`, use `.wrapper` and update usages from `<div id='wrapper'></div>` to `<div class="wrapper"></div>`. I'm sure you already knew how to do that, but just wanted to repeat it to show u that what you're doing is a habit and not a necessity.

mugayamadox commented:
> > Inorder to adjust, I need your recommendations on how I can improve on my DRY abilities and suggestions on a sample of what would be a better implementation. With that, I guess I fill the gaps.
> 
> Simple. Instead of an ID, use a CSS class. There's no mysterious knowledge to it. Your ID even have proper names that perfectly fit classes, so just use classes.
> 
> So, instead of `#wrapper`, use `.wrapper` and update usages from `<div id='wrapper'></div>` to `<div class="wrapper"></div>`. I'm sure you already knew how to do that, but just wanted to repeat it to show u that what you're doing is a habit and not a necessity.

Alright,
That clears all my inquiries.
Let me adjust my code

mugayamadox committed:
REFACTORING

REFACTORED code to replace some IDs with Classes
ADDED  styles to reset default browser styles

edwinmp commented on index.html:
```diff
@@ -6,13 +6,85 @@
-    It’s as yeezy as Kanye West. Search Know what you want to listen to? Just search and hit play. Browse Check out the
-    latest charts, brand new releases and great playlists for right now. Discover Enjoy new music every Monday with your
-    own personal playlist. Or sit back and enjoy Radio.
+    <div id="nav">
```
Make semantic and replace with a `nav` element instead of a div.

edwinmp commented on index.html:
```diff
@@ -6,13 +6,85 @@
+      </ul>
+    </div>
+
+    <div id="landing">
```
Demonstrate semantic HTML knowledge and replace these with `<section>` elements.

mugayamadox replied:
Ohkay
I can do that

edwinmp reviewed (changes requested):
We'll spare some time to review your implementation in our chat tomorrow so I can demonstrate DRY in action for your CSS.

mugayamadox committed:
REFACTORING: SEMANTICS

REFACTORED code to use SEMANTICS rather than DIVS

edwinmp commented:
@mugayamadox what's the status of the responsiveness?

mugayamadox committed:
IMPLEMENT:  Responsive Nav Bar

Added a responsive navigation bar with a collapsible menu on mobile phone screens.
